### PR TITLE
out_http: adding support for intermediate certificates (supersedes #3101)

### PR DESCRIPTION
**Pull request by tyarimi**
Signed-off-by: Tzach Yarimi <tyarimi@salesforce.com>

THIS PR REPLACES #3101 (switching the head branch)

**Which issue(s) this PR fixes**: 
Fixes #

**What this PR does / why we need it**: 
Add support for sending a client certificate chain for mutual TLS authentication in the `out_http` plugin.

**Docs Changes**:

**Release Note**: 
(same as PR name)

### CODE REVIEW

File: lib/fluent/plugin/out_http.rb [hunk 1]
```diff
@@ -21,6 +21,16 @@
 require 'fluent/plugin/output'
 require 'fluent/plugin_helper/socket'
 
+# patch Net::HTTP to support extra_chain_cert which was added in Ruby feature #9758.
+# see: https://github.com/ruby/ruby/commit/31af0dafba6d3769d2a39617c0dddedb97883712
+unless Net::HTTP::SSL_IVNAMES.include?(:@extra_chain_cert)
```
Comment (tyarimi) [inline lib/fluent/plugin/out_http.rb:26]: @nurse I added your suggested fix here (changed `SSL_IVNAMES` to `Net::HTTP::SSL_IVNAMES`)
Would love your review on the PR.

**Review by nurse — APPROVED** [review]

**Comment by nurse** [comment]
@repeatedly I'm ok about this

**Comment by repeatedly** [comment]
Thanks for the review and patch!

**Comment by ganga1980** [comment]
Hi, @nurse , wondering, were these changes related to setting intermediate proxy cert worked for you?

**Comment by zouyingjie** [comment]
Hi, @nurse @tyarimi  Dose have any examples for out_http mTLS configuration? I'm using out_http and configuring the configuration item of mTLS is always unsuccessful. Here is the issue https://github.com/fluent/fluentd/issues/3647.

**Comment by ganga1980** [comment]
> Hi, @nurse , wondering, were these changes related to setting intermediate proxy cert worked for you? I tested, these changes with my own http client but didn't work. if I pass the intermediate certs, TCP connection getting reset by the proxy server.

> @nurse, any updates on this?

**Pull request merged**